openupgrade_scripts/scripts/website_sale/14.0.1.0/post-migration.py

```diff
@@ -1,10 +1,51 @@
 # Copyright (C) 2021 Open Source Integrators <https://www.opensourceintegrators.com/>
 # Copyright 2021 ForgeFlow S.L.  <https://www.forgeflow.com>
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import re
 
 from openupgradelib import openupgrade
 
 
+def extract_custom_product_page_term_conditions(env):
+    """Replace Terms and Conditions content in the new v15 template so as not to lose
+    content from previous versions if it has been customised."""
+    product_views = env["ir.ui.view"].search(
+        [("key", "=", "website_sale.product"), ("website_id", "!=", False)]
+    )
+    for view in product_views:
+        product_arch = view.arch_db
+        product_pattern = r'<hr />\n<p class="text-muted">(.*?)<\/p>'
+        product_matches = re.findall(product_pattern, product_arch, re.DOTALL)
+        if product_matches:
+            standard_terms = """<hr />
+                <p class="text-muted">
+                30-day money-back guarantee<br />
```

Comment:
I'm afraid this doesn't serve, as this text may be translated by Odoo itself (if you load the website in the corresponding language). Let's write it always as a minor trade-off.

Reply:
I have removed it. In case the terms and conditions have not been modified a copy will be created as if they had been made, this is also not critical since if they are modified again in the future by a user no further copies of the view will be created, the changes will be reflected in the view with the associated website.
I can't think of anything universal to check since they are inserted directly into a view and simply create a copy with associated website when editing it but this is already covered in the search.